### PR TITLE
Rsx debug: Add some features

### PR DESCRIPTION
**Pull request by vlj**
This PR adds transform and shader programs associated to a draw call. It also dump the frame commands to command_dump.log.

It was used to find what were wrong with SH3 character models.

### CODE REVIEW

**Comment by vlj** [comment]
Fix build and rebased

**Comment by danilaml** [comment]
@vlj travis is same as before.

**Comment by vlj** [comment]
Rebased + fixed gcc build

**Comment by vlj** [comment]
Rebased

**Comment by Nekotekina** [comment]
Stop adding noexcept everywhere... It seems
1) Purposeless in most places
2) Even harmful sometimes

**Comment by vlj** [comment]
Why harmful sometimes ? 
noexcept is an hint telling the compiler that the function won't throw, and thus it can optimise away the seh branching logic in windows build. Plus containers like std::vector use a faster implementation if the underlying object has a non throwing move operator.

Of course it's harmful if a noexcept function is actually throwing but it should be fixed in the specific function, not ban noexcept for every other ones.

**Comment by Nekotekina** [comment]
What SEH branching logic? Anyway, you can't sacrifice correctness for some imaginary optimizations. So the safest way would be to avoid noexcept at all in the first place. If it's really necessary (some function is called from C callback, destructor or something else), it's different question.

**Comment by tambry** [comment]
@Nekotekina About optimizations, he probably means [this](http://stackoverflow.com/a/15461402/2176813).

**Comment by vlj** [comment]
Yet you don't tell how they can be harmful and want to sacrifice optimization for some imaginary uncorrectness claims.

**Comment by Nekotekina** [comment]
But you said it yourself: if the function actually throws, the exception is not passed and calls std::terminate. What should I add?

**Comment by danilaml** [comment]
@vlj this is a function though, not an constructor/destructor. So very little optimization could be made by compiler. And besides, SEH exceptions should be zero cost if they doesn't happen, same with noexcept and terminate. These functions are not truly noexcept too, there are possible OoM exceptions and such, though it shouldn't really matter.

File: rpcs3/Gui/RSXDebugger.h [hunk 1]
```diff
@@ -31,6 +30,9 @@ class RSXDebugger : public wxFrame
 	wxPanel* p_buffer_stencil;
 	wxPanel* p_buffer_tex;
 
+	wxTextCtrl* m_text_transform_program;
+	wxTextCtrl *m_text_shader_program;
```
Comment (danilaml) [inline rpcs3/Gui/RSXDebugger.h]: looks inconsistent.

**Comment by tambry** [comment]
@vlj Needs rebase.

**Comment by vlj** [comment]
Rebased + removed some noexcept

**Comment by vlj** [comment]
fixed build too

**Pull request merged**